### ISSUE TITLE
Authentication doesnt work if the app is embedded in a frame

### ISSUE THREAD

**Issue by ValentinKoch** [gradio-app/gradio#2472]
### Describe the bug

Pretty self explanatory, everything is working fine when I embed the gradio app in a div element except the login, which fails without any error message to the console of the browser.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Any gradio app that has authentication and is embedded as follows:
```
<html>
  <head>
  <meta name="keywords" content="" />
  <meta name="description" content="" />
  <meta http-equiv="content-type" content="text/html; charset=UTF-8" />
  </head>
  <frameset rows="100%">
  <frame src="http://18.195.21.198:7860/" title="mytitle" frameborder="0" noresize="noresize"/>
  <noframes>
  <body>
 <p><a href="http://mygradiohosting:7860">http://mywebsite.com</a>     </p>
  </body>
  </noframes>
  </frameset>
  </html>
```

### Logs

```shell
No logs in gradio nor in the browser console.
```

### System Info

```shell
Gradio 3.4, Google Chrome
```

### Severity

annoying

**Comment by abidlabs** [comment]
Thanks @ValentinKoch! @aliabid94 do you know if this is fixed by #2112?

**Comment by ValentinKoch** [comment]
Should add that i am hosting on aws ec2 maybe :) if that changes things

**Comment by leszekhanusz** [comment]
Could be related to #2266, probably a CORS issue.

**Comment by freddyaboulton** [comment]
I think this will be fixed by #3049 

I just replaced my space url (https://freddyaboulton-hello-login-main.hf.space/) in your document example and I was able to login.

You have to enable third party cookies however. 

![auth_iframe_embed](https://user-images.githubusercontent.com/41651716/214372780-75e9b9c8-7253-4a15-a3db-8e8780e47f3e.gif)
